### PR TITLE
endpointmetrics: Refactor manipulations on cluster monitoring config

### PR DESCRIPTION
**Pull request by philipgough**

### CODE REVIEW

File: operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go [hunk 1]
```diff
@@ -458,185 +409,182 @@ func createOrUpdateClusterMonitoringConfig(
...
+		foundClusterMonitoringConfiguration.PrometheusK8sConfig = newPmK8sConfig
 	}
-	found.Data[clusterMonitoringConfigDataKey] = string(updatedclusterMonitoringConfigurationYAMLBytes)
-	err = client.Update(ctx, found)
+
+	updatedClusterMonitoringConfigurationYAMLBytes, err := yaml.Marshal(foundClusterMonitoringConfiguration)
 	if err != nil {
-		log.Error(err, "failed to update configmap", "name", clusterMonitoringConfigName)
-		return err
+		log.Error(err, "failed to marshal the cluster monitoring config to yaml")
```
Comment (thibaultmg) [inline operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go:422]: You don't return an error here. Is it intentional?

File: operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go [hunk 2]
```diff
@@ -458,185 +409,182 @@ func createOrUpdateClusterMonitoringConfig(
...
+		return err
 	}
 
-	if !touched {
-		log.Info(
-			"endpoint-monitoring-operator is not a manager of configmap, no action needed",
-			"name",
-			clusterMonitoringConfigName,
-		)
+	log.Info("configmap exists, check if it needs revert", "name", clusterMonitoringConfigName)
```
Comment (thibaultmg) [inline operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go:441]: The CM does not necessarily exists at this point. Maybe update the message or change the logic?

File: operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go [hunk 3]
```diff
@@ -458,185 +409,182 @@ func createOrUpdateClusterMonitoringConfig(
...
-		log.Info("empty ClusterMonitoringConfiguration, should delete configmap", "name", clusterMonitoringConfigName)
+		log.Info("empty ClusterMonitoringConfiguration, deleting configmap if it still exists", "name", clusterMonitoringConfigName)
 		err = client.Delete(ctx, found)
-		if err != nil {
+		if err != nil && !errors.IsNotFound(err) {
 			log.Error(err, "failed to delete configmap", "name", clusterMonitoringConfigName)
 			return err
 		}
-		log.Info("configmap delete", "name", clusterMonitoringConfigName)
+		log.Info("configmap deleted", "name", clusterMonitoringConfigName)
```
Comment (thibaultmg) [inline operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go:498]: we could remove this log

**Review by thibaultmg — COMMENTED** [review]
Thanks for the refactoring, this looks better!
Left some minor comments for your consideration. The only potentially blocking one is about a missing return.

**Review by moadz — APPROVED** [review]
Nothing to add really, great job simplifying this mess :)

**Review by thibaultmg — APPROVED** [review]
lgtm

**Pull request merged**